### PR TITLE
protected Cpu info methods

### PR DESCRIPTION
**Pull request by Christian-B**
This PR looks at the Transceiver functions
get_cpu_information_from_cores and get_cpu_infos
to see why these are called.

Turns out the usages where
1. get_region_base_address
2. get_cores_in_states
3. get_cores_in_not_states
4. Print a String of the cores in/ not in states

So instead the methods that return detail CPUInfo objects have need protected and instead methods 1 to 3 have been added.

Methods 2 and 3 currently still return CPuInfo objects BUT only their iter method of this Object is used
This allows the Spin2 to do it differently if desired

Method 4 can actually be replicated by calling methods 2 or 3 using the x,y,p iter to a fill a set (or list) and then str that

There is a minor disadvantage that is rare cases 
ChipProvenanceUpdater if it is going to call an Exception then it needs to make a few extra calls 
Stand alone script get_cores_in_run_state also needs a few extra calls

Must be done at the same time as: 
https://github.com/SpiNNakerManchester/SpiNNFrontEndCommon/pull/1104
https://github.com/SpiNNakerManchester/sPyNNaker/pull/1379
https://github.com/SpiNNakerManchester/SpiNNakerGraphFrontEnd/pull/257

tested by:https://github.com/SpiNNakerManchester/IntegrationTests/pull/230

### CODE REVIEW

File: spinnman/transceiver.py [hunk 1]
```diff
@@ -918,9 +976,37 @@ def get_cpu_information_from_core(self, x, y, p):
...
+            If a response indicates an error during the exchange
+        """
+        a = self.read_user(x, y, p, 0)
+        core_subsets = CoreSubsets()
+        core_subsets.add_processor(x, y, p)
+        process = GetCPUInfoProcess(self._scamp_connection_selector)
+        cpu_info = process.get_cpu_info(core_subsets)
+        b = cpu_info.get_cpu_info(x, y, p).user[0]
+        assert a == b
+        return a
```
Comment (rowleya) [inline spinnman/transceiver.py:1008]: Seems odd to go to the trouble of doing this twice!

**Review by rowleya — CHANGES_REQUESTED** [review]
I think this isn't quite finished because of the assertion...

File: spinnman/transceiver.py (hunk 1 above)
Comment (Christian-B) [reply spinnman/transceiver.py:1008]: agree debug do it both ways. Removed

**Comment by Christian-B** [comment]
Replaced with https://github.com/SpiNNakerManchester/SpiNNMan/pull/366

**Pull request closed without merging**